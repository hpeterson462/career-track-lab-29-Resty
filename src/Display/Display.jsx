import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ text }) => (
  <div data-testid="display">
    {text}
  </div>
);

Display.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Display;
