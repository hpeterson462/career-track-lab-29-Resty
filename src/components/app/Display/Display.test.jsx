import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Display from './Display';

describe('Display component', () => {
  afterEach(() => cleanup());

  it('renders Display', () => {
    const { asFragment } = render(<Display
      text="Text Display"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
