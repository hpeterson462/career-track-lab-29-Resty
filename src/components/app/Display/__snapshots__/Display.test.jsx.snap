// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<DocumentFragment>
  <div
    data-testid="display"
  >
    Text Display
  </div>
</DocumentFragment>
`;
